src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -321,26 +356,69 @@ internal void RemoveProjectFromTrackingMaps_NoLock(ProjectId projectId)
 
         var project = Workspace.CurrentSolution.GetRequiredProject(projectId);
```

Comment:
can/should this line be removed

Reply:
yeah, updating.  this is used in the onAfterUpdate part.